poller/poller.go

```diff
@@ -136,16 +144,56 @@ func (poller *Poller) jobs() {
 	}
 }
 
+// getSerializedDataSubset returns serialized data for the splits provided
+func (poller *Poller) getSerializedDataSubset(splits []string) string {
```

Comment:
just to double confirm, so the first `serializedDataSubsets` in Cache will have non empty value only after someone calls `GetSerializedData` right? (not like `serializedData` which should have value after `poller.Start()` is called)

Reply:
That understanding is correct.